fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/InstrumentedTransaction.java

```diff
@@ -38,70 +39,74 @@
  * Wrapper around {@link Transaction} that instruments certain calls to expose their behavior with
  * {@link FDBStoreTimer} metrics.
  */
+@API(API.Status.INTERNAL)
 public class InstrumentedTransaction extends InstrumentedReadTransaction<Transaction> implements Transaction {
 
     @Nullable
     protected ReadTransaction snapshot; // lazily cached snapshot wrapper
 
-    public InstrumentedTransaction(@Nonnull StoreTimer timer, @Nonnull Transaction underlying) {
-        super(timer, underlying);
+    public InstrumentedTransaction(@Nonnull StoreTimer timer, @Nonnull Transaction underlying, boolean enableAssertions) {
+        super(timer, underlying, enableAssertions);
     }
 
     @Override
     public void addReadConflictRange(byte[] keyBegin, byte[] keyEnd) {
-        underlying.addReadConflictRange(keyBegin, keyEnd);
+        underlying.addReadConflictRange(checkKey(keyBegin), checkKey(keyEnd));
     }
 
     @Override
     public void addReadConflictKey(byte[] key) {
-        underlying.addReadConflictKey(key);
+        underlying.addReadConflictKey(checkKey(key));
     }
 
     @Override
     public void addWriteConflictRange(byte[] keyBegin, byte[] keyEnd) {
-        underlying.addWriteConflictRange(keyBegin, keyEnd);
+        underlying.addWriteConflictRange(checkKey(keyBegin), checkKey(keyEnd));
     }
 
     @Override
     public void addWriteConflictKey(byte[] key) {
-        underlying.addWriteConflictKey(key);
+        underlying.addWriteConflictKey(checkKey(key));
     }
 
     @Override
     public void set(byte[] key, byte[] value) {
-        underlying.set(key, value);
+        underlying.set(checkKey(key), checkValue(value));
         timer.increment(FDBStoreTimer.Counts.WRITES);
         timer.increment(FDBStoreTimer.Counts.BYTES_WRITTEN, key.length + value.length);
     }
 
     @Override
     public void clear(byte[] key) {
-        underlying.clear(key);
+        underlying.clear(checkKey(key));
         timer.increment(FDBStoreTimer.Counts.DELETES);
     }
 
     @Override
     public void clear(byte[] keyBegin, byte[] keyEnd) {
-        underlying.clear(keyBegin, keyEnd);
+        underlying.clear(checkKey(keyBegin), checkKey(keyEnd));
         timer.increment(FDBStoreTimer.Counts.DELETES);
     }
 
     @Override
     public void clear(Range range) {
+        checkKey(range.begin);
+        checkKey(range.end);
+
         underlying.clear(range);
         timer.increment(FDBStoreTimer.Counts.DELETES);
     }
 
     @Override
     @Deprecated
     public void clearRangeStartsWith(byte[] prefix) {
-        underlying.clearRangeStartsWith(prefix);
+        underlying.clearRangeStartsWith(checkKey(prefix));
         timer.increment(FDBStoreTimer.Counts.DELETES);
     }
 
     @Override
     public void mutate(MutationType opType, byte[] key, byte[] param) {
-        underlying.mutate(opType, key, param);
+        underlying.mutate(opType, checkKey(key), param);
```

Comment:
```suggestion
        underlying.mutate(opType, checkKey(key), checkValue(param));
```

Reply:
Technically, the versionstamp operations can _shrink_ the key and value sizes, and...I'm not actually sure whether a 100,004 byte value where the last four bytes are a versionstamp offset as part of a `SET_VERSIONSTAMPED_VALUE` operation are actually allowed.